app/controllers/journeys/expenses/tailoring/individualCategories/FinancialExpensesController.scala

```diff
@@ -79,21 +83,37 @@ class FinancialExpensesController @Inject() (override val messagesApi: MessagesA
 
   def onSubmit(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) async {
     implicit request =>
-      selfEmploymentService.getAccountingType(request.user.nino, businessId, request.user.mtditid) flatMap {
-        case Left(_) => Future.successful(Redirect(JourneyRecoveryController.onPageLoad()))
-        case Right(accountingType) =>
-          formProvider(request.userType)
-            .bindFromRequest()
-            .fold(
-              formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, request.userType, taxYear, businessId, accountingType))),
-              value =>
-                for {
-                  clearedAnswers <- Future.fromTry(clearPageDataFromUserAnswers(request.userAnswers, Some(businessId), value))
-                  updatedAnswers <- Future.fromTry(clearedAnswers.set(FinancialExpensesPage, value, Some(businessId)))
-                  _              <- sessionRepository.set(updatedAnswers)
-                } yield Redirect(navigator.nextPage(FinancialExpensesPage, mode, updatedAnswers, taxYear, businessId))
-            )
-      }
+      def handleError(formWithErrors: Form[_], userType: UserType, accountingType: AccountingType): Future[Result] =
+        Future.successful(
+          BadRequest(view(formWithErrors, mode, userType, taxYear, businessId, accountingType))
+        )
+
+      def handleSuccess(userAnswers: UserAnswers, answers: Set[FinancialExpenses]): Future[Result] =
```

Comment:
Could we maybe rename answers somehow to something a bit more distinct e.g. `financialExpenses` or wtte? Lots of ..answers here, `clearedAnswers`, `answers`, `userAnswers` etc.